src/App.js

```diff
@@ -32,16 +32,37 @@ const darkTheme = createMuiTheme( {
   tonalOffset: 0.2,
 } )
 
+const commitSettings = settings => {
+  Object.entries( settings ).forEach( ( [ name, value ] ) => {
+    const { storageKey } = OPTIONS[name]
+    writeCss( storageKey, value )
+    window.localStorage.setItem( storageKey, value )
+  } )
+}
+
+const applyLockPadding = ( settings, updatedSettings ) => {
+  const { horizontalPadding } = updatedSettings
+  const { lockOverlayPadding } = { ...settings, ...updatedSettings }
+
+  if ( !lockOverlayPadding ) return updatedSettings
+
+  const verticalPadding = horizontalPadding || settings.horizontalPadding
+  commitSettings( { verticalPadding } )
+
+  return {
+    ...updatedSettings,
+    // Sync the sliders up but using the same numeric value by giving it the wrong unit
```

Comment:
Can we use the [Material Slider](https://material-ui.com/api/slider/#props) `disabled` prop?

Reply:
Would it be better/more useful? Certainly can do, but not too sure on how to feed that down. Maybe it could be if the value is `false`, then the slider is disabled.

Reply:
I added `disabled:false` in `options.js` and modified the reducer it works but needs cleaning 
```
const applyLockPadding = ( settings, updatedSettings ) => {
  const { horizontalPadding } = updatedSettings
  const { lockOverlayPadding } = { ...settings, ...updatedSettings }

  // if ( !lockOverlayPadding ) return updatedSettings
  if ( !lockOverlayPadding ) {
    OPTIONS.verticalPadding.disabled = false
    return updatedSettings
  }

  const verticalPadding = horizontalPadding || settings.horizontalPadding
  commitSettings( { verticalPadding } )
  OPTIONS.verticalPadding.disabled = true
  return {
    ...updatedSettings,
    // Sync the sliders up but using the same numeric value by giving it the wrong unit
    verticalPadding: verticalPadding.replace( OPTIONS.horizontalPadding.units, OPTIONS.verticalPadding.units ),
  }
}
```

Reply:
`OPTIONS.verticalPadding.disabled = !lockOverlayPadding` above the return

Reply:
Make a note that this is a hack, since React does not know about this and we are relying on race condition

Reply:
Is there any other way ?

Reply:
Not anything particularly satiable, except moving the options to the state. But this should be fine for the use-case, just don't like it particularly much.

Reply:
Done. Take a look at how I've done it.